extern/guided_diffusion/guided_diffusion/train_util.py

```diff
@@ -33,15 +40,20 @@ def __init__(
         log_interval,
         save_interval,
         resume_checkpoint,
+        val_data=None,
         use_fp16=False,
         fp16_scale_growth=1e-3,
         schedule_sampler=None,
         weight_decay=0.0,
         lr_anneal_steps=0,
+        target: str=None,
+        save_opt: bool=True,
+        save_ema: bool=True
     ):
         self.model = model
         self.diffusion = diffusion
         self.data = data
+        self.val_data: DataLoader = val_data
```

Comment:
here aswell

Reply:
This one makes sense since data and val_data are related